sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -294,22 +293,18 @@ public void withScope(@NotNull ScopeCallback callback) {
           "Instance is disabled and this 'withScope' call is a no-op.");
     } else {
       pushScope();
-      try {
-        StackItem item = stack.peek();
-        if (item != null) {
-          try {
-            callback.run(item.scope);
-          } catch (Exception e) {
-            logIfNotNull(
-                options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
-          }
-        } else {
+      StackItem item = stack.peek();
+      if (item != null) {
+        try {
+          callback.run(item.scope);
+        } catch (Exception e) {
           logIfNotNull(
-              options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
+              options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
         }
-      } finally {
-        popScope();
+      } else {
+        logIfNotNull(options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
```

Comment:
What is the purpose of using `logIfNotNull` with `options.getLogger()` returns a `@NonNull` value?

Reply:
answered here https://github.com/getsentry/sentry-android/pull/96#discussion_r342470530